File: examples/UnclaimedDraftCreateEmbeddedFormFieldGroupsExample.rb
require "json"
require "dropbox-sign"

Dropbox::Sign.configure do |config|
    config.username = "YOUR_API_KEY"
    # config.access_token = "YOUR_ACCESS_TOKEN"
end

form_field_groups_1 = Dropbox::Sign::SubFormFieldGroup.new
form_field_groups_1.group_id = "RadioItemGroup1"
form_field_groups_1.group_label = "Radio Item Group 1"
form_field_groups_1.requirement = "require_0-1"

form_field_groups = [
    form_field_groups_1,
]

form_fields_per_document_1 = Dropbox::Sign::SubFormFieldsPerDocumentRadio.new
form_fields_per_document_1.document_index = 0
form_fields_per_document_1.api_id = "uniqueIdHere_1"
form_fields_per_document_1.type = "radio"
form_fields_per_document_1.required = false
form_fields_per_document_1.signer = "0"
form_fields_per_document_1.width = 18
form_fields_per_document_1.height = 18
form_fields_per_document_1.x = 112
form_fields_per_document_1.y = 328
form_fields_per_document_1.group = "RadioItemGroup1"
form_fields_per_document_1.is_checked = true
form_fields_per_document_1.name = ""
form_fields_per_document_1.page = 1

form_fields_per_document_2 = Dropbox::Sign::SubFormFieldsPerDocumentRadio.new
form_fields_per_document_2.document_index = 0
form_fields_per_document_2.api_id = "uniqueIdHere_2"
form_fields_per_document_2.type = "radio"
form_fields_per_document_2.required = false
form_fields_per_document_2.signer = "0"
form_fields_per_document_2.width = 18
form_fields_per_document_2.height = 18
form_fields_per_document_2.x = 112
form_fields_per_document_2.y = 370
form_fields_per_document_2.group = "RadioItemGroup1"
form_fields_per_document_2.is_checked = false
form_fields_per_document_2.name = ""
form_fields_per_document_2.page = 1

form_fields_per_document = [
    form_fields_per_document_1,
    form_fields_per_document_2,
]

unclaimed_draft_create_embedded_request = Dropbox::Sign::UnclaimedDraftCreateEmbeddedRequest.new
unclaimed_draft_create_embedded_request.client_id = "b6b8e7deaf8f0b95c029dca049356d4a2cf9710a"
unclaimed_draft_create_embedded_request.requester_email_address = "jack@dropboxsign.com"
unclaimed_draft_create_embedded_request.test_mode = false
unclaimed_draft_create_embedded_request.file_urls = [
    "https://www.dropbox.com/s/ad9qnhbrjjn64tu/mutual-NDA-example.pdf?dl=1",
]
unclaimed_draft_create_embedded_request.form_field_groups = form_field_groups
unclaimed_draft_create_embedded_request.form_fields_per_document = form_fields_per_document

begin
    response = Dropbox::Sign::UnclaimedDraftApi.new.unclaimed_draft_create_embedded(
        unclaimed_draft_create_embedded_request,
    )

    p response
rescue Dropbox::Sign::ApiError => e
    puts "Exception when calling UnclaimedDraftApi#unclaimed_draft_create_embedded: #{e}"
end

File: examples/UnclaimedDraftCreateEmbeddedFormFieldGroupsExample.java
package com.dropbox.sign_sandbox;

import com.dropbox.sign.ApiException;
import com.dropbox.sign.Configuration;
import com.dropbox.sign.api.*;
import com.dropbox.sign.auth.*;
import com.dropbox.sign.JSON;
import com.dropbox.sign.model.*;

import java.io.File;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class UnclaimedDraftCreateEmbeddedFormFieldGroupsExample
{
    public static void main(String[] args)
    {
        var config = Configuration.getDefaultApiClient();
        ((HttpBasicAuth) config.getAuthentication("api_key")).setUsername("YOUR_API_KEY");
        // ((HttpBearerAuth) config.getAuthentication("oauth2")).setBearerToken("YOUR_ACCESS_TOKEN");

        var formFieldGroups1 = new SubFormFieldGroup();
        formFieldGroups1.groupId("RadioItemGroup1");
        formFieldGroups1.groupLabel("Radio Item Group 1");
        formFieldGroups1.requirement("require_0-1");

        var formFieldGroups = new ArrayList<SubFormFieldGroup>(List.of (
            formFieldGroups1
        ));

        var formFieldsPerDocument1 = new SubFormFieldsPerDocumentRadio();
        formFieldsPerDocument1.documentIndex(0);
        formFieldsPerDocument1.apiId("uniqueIdHere_1");
        formFieldsPerDocument1.type("radio");
        formFieldsPerDocument1.required(false);
        formFieldsPerDocument1.signer("0");
        formFieldsPerDocument1.width(18);
        formFieldsPerDocument1.height(18);
        formFieldsPerDocument1.x(112);
        formFieldsPerDocument1.y(328);
        formFieldsPerDocument1.group("RadioItemGroup1");
        formFieldsPerDocument1.isChecked(true);
        formFieldsPerDocument1.name("");
        formFieldsPerDocument1.page(1);

        var formFieldsPerDocument2 = new SubFormFieldsPerDocumentRadio();
        formFieldsPerDocument2.documentIndex(0);
        formFieldsPerDocument2.apiId("uniqueIdHere_2");
        formFieldsPerDocument2.type("radio");
        formFieldsPerDocument2.required(false);
        formFieldsPerDocument2.signer("0");
        formFieldsPerDocument2.width(18);
        formFieldsPerDocument2.height(18);
        formFieldsPerDocument2.x(112);
        formFieldsPerDocument2.y(370);
        formFieldsPerDocument2.group("RadioItemGroup1");
        formFieldsPerDocument2.isChecked(false);
        formFieldsPerDocument2.name("");
        formFieldsPerDocument2.page(1);

        var formFieldsPerDocument = new ArrayList<SubFormFieldsPerDocumentBase>(List.of (
            formFieldsPerDocument1,
            formFieldsPerDocument2
        ));

        var unclaimedDraftCreateEmbeddedRequest = new UnclaimedDraftCreateEmbeddedRequest();
        unclaimedDraftCreateEmbeddedRequest.clientId("b6b8e7deaf8f0b95c029dca049356d4a2cf9710a");
        unclaimedDraftCreateEmbeddedRequest.requesterEmailAddress("jack@dropboxsign.com");
        unclaimedDraftCreateEmbeddedRequest.testMode(false);
        unclaimedDraftCreateEmbeddedRequest.fileUrls(List.of (
            "https://www.dropbox.com/s/ad9qnhbrjjn64tu/mutual-NDA-example.pdf?dl=1"
        ));
        unclaimedDraftCreateEmbeddedRequest.formFieldGroups(formFieldGroups);
        unclaimedDraftCreateEmbeddedRequest.formFieldsPerDocument(formFieldsPerDocument);

        try
        {
            var response = new UnclaimedDraftApi(config).unclaimedDraftCreateEmbedded(
                unclaimedDraftCreateEmbeddedRequest
            );

            System.out.println(response);
        } catch (ApiException e) {
            System.err.println("Exception when calling UnclaimedDraftApi#unclaimedDraftCreateEmbedded");
            System.err.println("Status code: " + e.getCode());
            System.err.println("Reason: " + e.getResponseBody());
            System.err.println("Response headers: " + e.getResponseHeaders());
            e.printStackTrace();
        }
    }
}
